source/tests/system/nirfmxnr_session_tests.cpp

```diff
@@ -0,0 +1,119 @@
+#include <gtest/gtest.h>
+
+#include "device_server.h"
+#include "nirfmxnr/nirfmxnr_service.h"
+
+namespace ni {
+namespace tests {
+namespace system {
+namespace {
+
+namespace rfmxnr = nirfmxnr_grpc;
+
+const int kInvalidRsrc = -200220;
+const int kInvalidRFmxNRSession = -380598;
+const char* kRFmxNRTestRsrc = "FakeDevice";
+const char* kRFmxNROptionsString = "Simulate=1, DriverSetup=Model:5663E";
+const char* kRFmxNRTestSession = "SessionName";
+const char* kRFmxNRTestInvalidRsrc = "";
+
+class NiRFmxNRSessionTest : public ::testing::Test {
+ protected:
+  NiRFmxNRSessionTest()
+      : device_server_(DeviceServerInterface::Singleton()),
+        nirfmxnr_stub_(rfmxnr::NiRFmxNR::NewStub(device_server_->InProcessChannel()))
+  {
+    device_server_->ResetServer();
+  }
+
+  virtual ~NiRFmxNRSessionTest() {}
+
+  std::unique_ptr<rfmxnr::NiRFmxNR::Stub>& GetStub()
+  {
+    return nirfmxnr_stub_;
+  }
+
+  ::grpc::Status call_initialize(const char* resource_name, const char* option_string, const char* session_name, rfmxnr::InitializeResponse* response)
+  {
+    ::grpc::ClientContext context;
+    rfmxnr::InitializeRequest request;
+    request.set_resource_name(resource_name);
+    request.set_option_string(option_string);
+    request.set_session_name(session_name);
+
+    ::grpc::Status status = GetStub()->Initialize(&context, request, response);
+    return status;
+  }
+
+ private:
+  DeviceServerInterface* device_server_;
+  std::unique_ptr<rfmxnr::NiRFmxNR::Stub> nirfmxnr_stub_;
+};
+
+TEST_F(NiRFmxNRSessionTest, InitializeSessionWithDeviceAndSessionName_CreatesDriverSession)
```

Comment:
I made a comment on your work item, but we should capture these tests in the Acceptance table.

I also would like to test initializing a 2nd session to the same device